cloud/azure/bin/lib/azure.sh

```diff
@@ -168,3 +168,26 @@ function azure::swap_deployment_slot() {
 function azure::get_current_user_id() {
   az ad signed-in-user show --query mail | sed -E 's/ +/_/g'
 }
+
+#######################################
+# Swap the canary slot into production.
+# Arguments:
+#   1. The resource group name
+#   2. role guid
+#   3. scope name
+#######################################
+function azure::create_role_assignment() {
```

Comment:
suggestion (blocking): recommend renaming to `ensure_role_assignment` since it does the check and bails if they already have it